### PR TITLE
Fix: Fixed an issue with drag and dropping items onto .ahk files

### LINKED ISSUE #14430: Bug: Unable to drag and drop any file onto .ahk files
### Description

The drag-and-drop functionality seems to have a limitation, preventing the dragging and dropping of files with any file extension onto .ahk files.

I found a similar closed issue with .py files but that one has been fixed. Dragging and dropping doesn't seem to be an option.

### Steps To Reproduce

Make an .ahk file
Try to drag any file onto it.

### Requirements

It should allow me to drag and drop file onto .ahk files, but it currently does not. Essentially run that .ahk file with the filepath(s) as a parameter 

### Files Version

3.1.1.0

### Windows Version

10.0.22631 Build 22631

### Log File

[debug.log](https://github.com/files-community/Files/files/13903822/debug.log)

### PR DESCRIPTION
**Pull request by lunathanael**
Allow .ahk and .py files to be treated as executable files to allow dragging and dropping other files onto them.

This was accomplished by essentially replacing current support for Python to also include AutoHotkey by combining the Script category.

Removed checks for Python/AHK Installed as AHK (and potentially other scripts) does not support version checking. This can be replaced by an alternative method for checking interpreter installation.

~Note: AutoHotkey.exe is required to be added to the path, there is currently no check for this.~

**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #14430 

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [x] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?
   1. Open app ...
   2. Click settings button ...

**Screenshots (optional)**
image
image

### CODE REVIEW

**Review by 0x5bfa — COMMENTED** [review]
Thanks for the first time contribution!

> Note: AutoHotkey.exe is required to be added to the path, there is currently no check for this.

Does this mean that execution continues still the app doesn't check if python is installed? If so, is that ok?

**Comment by lunathanael** [comment]
> Thanks for the first time contribution!
> 
> > Note: AutoHotkey.exe is required to be added to the path, there is currently no check for this.
> 
> Does this mean that execution continues still the app doesn't check if python is installed? If so, is that ok?

Yes, the execution will continue.
Correct me if I'm wrong, but when you open a file, it attempts to use the "default" application to do so, and when none is defined it'll prompt for one. For example, if I open a .ahk file without AutoHotkey installed and added to PATH, it'll prompt for a method to open the script, as if you pressed "open with...".

Thank you for clarifying this, just trying out something relatively easy as my first PR, definitely made some mistakes.

**Comment by 0x5bfa** [comment]
Thanks for the explanation!
You chose a right one:)

Don't worry, I will review on my local and approve.

**Comment by lunathanael** [comment]
> Thanks for the explanation! You chose a right one:)
> 
> Don't worry, I will review on my local and approve.

Thanks for the encouragement.
I just tested again and I believe that there is no need for AutoHotkey to be in the PATH.
My initial thought seem to be correct.
image

File: src/Files.App/Utils/Storage/Operations/FilesystemHelpers.cs [hunk 1]
```diff
@@ -256,11 +256,9 @@ public async Task<ReturnResult> RestoreItemsFromTrashAsync(IEnumerable<IStorageI
 				else if (operation.HasFlag(DataPackageOperation.Link))
 				{
 					// Open with piggybacks off of the link operation, since there isn't one for it
-					if (isTargetExecutable || isTargetPythonFile)
+					if (isTargetExecutable || isTargetScriptFile)
 					{
 						var items = await GetDraggedStorageItems(packageView);
-						if (isTargetPythonFile && !SoftwareHelpers.IsPythonInstalled())
```
Comment (yaira2) [inline src/Files.App/Utils/Storage/Operations/FilesystemHelpers.cs:262]: This check is still needed to make sure the target software is available

**Comment by 0x5bfa** [comment]
And I wanna merge this too until the next release. @yaira2

File: src/Files.App/Utils/Storage/Operations/FilesystemHelpers.cs (hunk 1 above)
Comment (yaira2) [inline src/Files.App/Utils/Storage/Operations/FilesystemHelpers.cs:262]: What happens if the target isn't installed?

Comment (lunathanael) [reply src/Files.App/Utils/Storage/Operations/FilesystemHelpers.cs:262]: https://github.com/files-community/Files/pull/14895#issuecomment-1975617972

Comment (lunathanael) [reply src/Files.App/Utils/Storage/Operations/FilesystemHelpers.cs:262]: Looking back, I think the comment was unclear. There is no requirement for AutoHotkey to be in the PATH. The program will still function as intended. This picture is if AutoHotkey is neither installed nor in the PATH.

Comment (yaira2) [reply src/Files.App/Utils/Storage/Operations/FilesystemHelpers.cs:262]: I meant, what happens if Python isn't installed?

Comment (lunathanael) [reply src/Files.App/Utils/Storage/Operations/FilesystemHelpers.cs:262]: From my testing, it treats the file as if you would normally try to open a .py without python installed. If no windows default application for opening the file type is defined, it'll prompt for one like the picture suggests. I'm not sure if I understand or have answered your question correctly, I apologize.

**Review by yaira2 — APPROVED** [review]
LGTM

**Comment by yaira2** [comment]
@lunathanael thank you for contributing to Files!

**Pull request merged**